07-data_pipelines.Rmd

```diff
@@ -80,6 +80,75 @@ The repo should include:
+
+This will create an `airflow` directory in your home directory on the Analytical Platform. In this airflow directory there are three more directories `db`, `dags` and `logs`. The `dags` directory is where you will store your test DAG files.
+
+### Create a test DAG
+
+Using JupyterLab on the Analytical Platform create a python file in the `airflow/dags` directory in your home directory in the Analytical Platform (eg. `test_dag.py`). Airflow will automatically scan this directory for DAG files every three minutes.
+
+See above for creating a Docker image to run the DAG if you are using `KubernetesPodOperator`. Concourse will automatically build images of PR's to GitHub Airflow repos and will be tagged with the name of the PR.
+
+Example DAG file contents:
+
+
+    from datetime import datetime
+
+    from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+    from airflow.utils.log.logging_mixin import LoggingMixin
+    from airflow.models import DAG
+
+    log = LoggingMixin().log
+
+    args = {"owner": "airflow", "start_date": datetime(2019, 3, 18)}
+
+    dag = DAG(
+        dag_id="example_kubernetes_operator_assume_role",
+        default_args=args,
+        schedule_interval=None,
+    )
+
+    k = KubernetesPodOperator(
+        namespace="user-###YOUR_GITHUB_USERNAME###",
+        image="governmentpaas/awscli:latest",
+        cmds=["aws", "--debug"],
+        arguments=["s3api", "list-objects", "--bucket", "###YOUR_TEST_BUCKET###"],
+        env_vars={
+            "AWS_METADATA_SERVICE_TIMEOUT": "60",
+            "AWS_METADATA_SERVICE_NUM_ATTEMPTS": "5",
+        },
+        labels={"foo": "bar"},
+        name="airflow-test-pod",
+        in_cluster=True,
+        task_id="task",
+        get_logs=True,
+        dag=dag,
+        is_delete_operator_pod=True,
+        annotations={"iam.amazonaws.com/role": "alpha_user_###YOUR_GITHUB_USERNAME###"},
+    )
+
+
+
+You will have to set the DAG namespace to your namespace and the AWS role to your role, see `###YOUR_GITHUB_USERNAME###` in the above example code.
+
+You will also need to update the image with your DAG image from AWS ECR and add any environment variables etc. as you would with a normal DAG.
+
+### IMPORTANT
+
+This sandbox Airflow is meant for testing. You should not use it to run regular tasks and should not be relied on for running production pipelines. It is likely that the sandbox will be idled when it is not being used.
+
+Because you are running this DAG as your own `alpha_user_...`, the DAG will have the access that your user has. It does not run with the `iam_policy.json` that you may have created in the task GitHub repo. This means that you might need to either grant yourself access to any buckets or copy data in to a bucket you have created in the Warehouse data section of the Analytical Platform.
```

Comment:
```suggestion
Because DAGs are run as your own `alpha_user_...`, they will have the same access permissions. DAGs are not run using an `iam_policy.json`, even if one exists in the task GitHub repository. This means that you may need to either obtain access to any required buckets or copy data into a bucket that you already have access to.
```

Reply:
:+1:

Reply:
I reckon this might usefully have another sentence on @aldo's point i.e. to warn that your role can do potentially do bad things, with likely access to buckets across all your projects.